### PR TITLE
Clean up legacy code for document collections

### PR DESCRIPTION
**Pull request by mtaylorgds**
## What

Remove vestigial code that has been left behind after the migration of the document collection pages to the GOV.UK Design System.

## Trello

[Trello card](https://trello.com/c/dNqJdXD0).

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: features/document-collections.feature
Comment (farahTW) [inline features/document-collections.feature:1]: Just one query on this feature test, are we supposed to remove these feature tests?
or just remove anootations @javascript @design-system-wip?

**Comment by davidtrussler** [comment]
I'm not sure if we are planning to remove all files under `javascripts/admin_legacy` and `stylesheets/admin_legacy` in one go, though to date the process has been to remove JavaScript and CSS files as and when they become surplus to requirements, which is probably the netter way to go IMO. 

If that is the case then we should also delete these files at this point:
- app/assets/javascripts/admin_legacy/_document_group_ordering.js
- app/assets/javascripts/admin_legacy/document_finder.js
- app/assets/javascripts/admin_legacy/document_collection.js
- app/assets/stylesheets/admin_legacy/views/_document-collection-groups.scss
- app/assets/stylesheets/admin_legacy/helpers/_document-finder.scss

**Comment by davidtrussler** [comment]
Just a query on the status of this, and one other, PR: is this one based on [PR#8415](https://github.com/alphagov/whitehall/pull/8415) ("Prepare release for document collections") and can that one therefore be closed? It does seem to be an extended duplicate of that work.

**Comment by mtaylorgds** [comment]
> Just a query on the status of this, and one other, PR: is this one based on [PR#8415](https://github.com/alphagov/whitehall/pull/8415) ("Prepare release for document collections") and can that one therefore be closed? This seems to be an extended duplicate of that work.

By necessity, this PR had to be branched off of [PR#8415](https://github.com/alphagov/whitehall/pull/8415) (since we can't cleanup until that work has been merged). The intention is that once that PR is merged this one can be rebased off of the latest main, and so will only contain the "remove" commits.

**Comment by mtaylorgds** [comment]
> I'm not sure if we are planning to remove all files under `javascripts/admin_legacy` and `stylesheets/admin_legacy` in one go, though to date the process has been to remove JavaScript and CSS files as and when they become surplus to requirements, which is probably the better way to go IMO.
> 
> If that is the case then we should also delete these files at this point:
> 
> * app/assets/javascripts/admin_legacy/_document_group_ordering.js
> * app/assets/javascripts/admin_legacy/document_finder.js
> * app/assets/javascripts/admin_legacy/document_collection.js
> * app/assets/stylesheets/admin_legacy/views/_document-collection-groups.scss
> * app/assets/stylesheets/admin_legacy/helpers/_document-finder.scss

It would be nice to remove unnecessary files as and when we go (ultimately leading to the removal of everything under `javascripts/admin_legacy` and  `stylesheets/admin_legacy`).

In this case those files are also used by some files under `statistics_announcements`, which I believe are ultimately not used anymore, but given the complexity of so many interdependencies I thought it best to do as a separate ticket—see the "[cleanup document_searches code](https://trello.com/c/wvra2ses)" Trello card.

**Review by RodneyJohnsonGDS — APPROVED** [review]
LGTM!

**Pull request merged**